code/game/objects/effects/mines.dm

```diff
@@ -146,10 +146,13 @@
 	chainsaw.wield(victim)
 	victim.reagents.add_reagent("adminordrazine", 25)
 
-	victim.client.color = pure_red
-	animate(victim.client,color = red_splash, time = 10, easing = SINE_EASING|EASE_OUT)
-	spawn(10)
-		animate(victim.client,color = old_color, time = duration)//, easing = SINE_EASING|EASE_OUT)
+	// Only mess with their screen colour if they arent IRL colourblind
+	if(victim.client?.prefs.colourblind_mode == COLOURBLIND_MODE_NONE)
+		victim.client.color = pure_red
+		animate(victim.client,color = red_splash, time = 10, easing = SINE_EASING|EASE_OUT)
+		spawn(10)
+			animate(victim.client,color = old_color, time = duration)//, easing = SINE_EASING|EASE_OUT)
```

Comment:
would you convert this into a general proc? that way if anyone creates a similar effect they don't have to remember to make an exception for the colourblind.

also would you convert the spawn to a timer?

Reply:
I am 99% sure we have a general proc for this anyway, the `flash_screen_colour` one, so I should just use that and not the spawn anyways.

Reply:
See 80ae8047ce1783c89e880e24e99b6990626bf17d